### PR TITLE
[Epinio] Adapt the UI to the new manifest API

### PR DESCRIPTION
**Pull request by torchiaf**
### Summary
Fixes https://github.com/epinio/ui/issues/228
Fixes https://github.com/epinio/ui/issues/227

### CODE REVIEW

**Review by richard-cox — CHANGES_REQUESTED** [review]
Haven't given this a full functional test yet, will save that until after EPINIO_APP_DATA has been removed. 

I did test a succesful deploy from github and download manifest. The manifest however, when loaded in the create app process, thinks it's git instead of github

File: pkg/epinio/detail/applications.vue [hunk 1]
```diff
@@ -19,7 +17,7 @@ import AppGitDeployment from '../components/application/AppGitDeployment.vue';
 import Link from '@shell/components/formatter/Link.vue';
 import { GitUtils, gitUtilsToLabel } from '../utils/git';
 import { isArray } from '@shell/utils/array';
-import { ucFirst } from '@shell/utils/string';
+import { AppUtils } from '@/pkg/epinio/utils/application';
```
[the remark this answers is not in the document]
Comment (torchiaf) [reply pkg/epinio/detail/applications.vue:20]: Using `@pkg` breaks the app; is is also marked as error in VS code

File: pkg/epinio/detail/applications.vue [hunk 2]
```diff
@@ -67,7 +68,8 @@ export default Vue.extend<Data, any, any, any>({
     return {
       APPLICATION_MANIFEST_SOURCE_TYPE,
       saving:        false,
-      envs:          this.value?.envDetails,
+      source:        AppUtils.getAppSource(this.value),
```
[the remark this answers is not in the document]
Comment (torchiaf) [reply pkg/epinio/detail/applications.vue:71]: Agree, I'm going to move `getAppSource` and `getAppSourceInfo` in the applications model and leave the other utils functions in `application.ts` util.

File: pkg/epinio/types.ts [hunk 3]
```diff
@@ -91,24 +91,23 @@ export interface EpinioAppSource {
...
-export const APPLICATION_ENV_VAR = 'EPINIO_APP_DATA';
-
-export type EPINIO_APP_ENV_VAR_GIT = Omit<AppSourceGit, 'sourceData'>;
+export type EPINIO_APP_GIT_SOURCE = Omit<AppSourceGit, 'sourceData'>;
 
 /**
  * Contains persisted information that provides applications an understanding of where they came from
  */
+// Todo Use it
 export interface EPINIO_APP_DATA {
```
[the remark this answers is not in the document]
Comment (torchiaf) [reply pkg/epinio/types.ts:100]: Yes, good point. I need to a refactoring.

**Comment by torchiaf** [comment]
Thanks @richard-cox , I updated the code with the requested changes.
> will save that until after EPINIO_APP_DATA has been removed.
The EPINIO_APP_DATA is removed in this PR, so ti will resolve both issues in the PR description. Am I missing somenthing ?
> The manifest however, when loaded in the create app process, thinks it's git instead of github
We have this bug also in `epinio-dev`, so could we open a new issue for this ?

File: pkg/epinio/edit/applications.vue [hunk 4]
```diff
@@ -232,6 +220,7 @@ export default Vue.extend<Data, any, any, any>({
         >
           <template #source>
             <AppSource
+              v-if="source"
```
Comment (richard-cox) [inline pkg/epinio/edit/applications.vue:223]: for neatness, can this be tied in to `showSourceTab`?

File: pkg/epinio/models/application-action.js [hunk 5]
```diff
@@ -154,9 +148,15 @@ export default class ApplicationActionResource extends Resource {
     this.application.showAppLog();
   }
 
+  // Todo move to utils
```
Comment (richard-cox) [inline pkg/epinio/models/application-action.js:151]: ToDo (though not sure we need to move this outside of the file)

**Review by richard-cox — APPROVED** [review]
Code reviewed, however blocked from functional testing due to the git provider type changes brought in when our epinio instance was rebuilt. Will aim to get all PRs merged,  then test, then move issues to To Test.

With that in mind, we can ignore the two comments in this review

**Pull request merged**